### PR TITLE
feat(CRT-143): add KubeFedCluster CRD & controller

### PR DESCRIPTION
**Pull request by MatousJobanek**
Integrates KubeFedCluster CRD and related controller to the Member operator.
Changes proposed as part of this PR:
* use the latest released version of KubeFed project `v0.1.0-rc2`
* add `core_v1beta1_kubefedcluster.yaml` CRD that is just generated CRD from the KubeFed of the version `v0.1.0-rc2` 
* change role of the SA to clusterrole as it is required by KubeFedCluster (and add "list nodes" permission)
* uses script `add-cluster.sh` from toolchain repository: https://github.com/codeready-toolchain/toolchain/pull/2
* imports & starts KubeFedClusterController

### CODE REVIEW

File: make/dev.mk [hunk 1]
```diff
@@ -10,7 +10,68 @@ IMAGE_NAME?=member-operator
...
-# ToDo: add required make target to build and run operator in dev environment later once we have openshift-ci in place.
+# to watch all namespaces, keep namespace empty
+APP_NAMESPACE ?= ""
+LOCAL_TEST_NAMESPACE ?= "test-saas-control-plane"
 
+.PHONY: local
+## Run Operator locally
+local: login-as-admin create-namespace deploy-rbac build deploy-crd
+	$(Q)-oc new-project $(LOCAL_TEST_NAMESPACE) || true
+	$(Q)OPERATOR_NAMESPACE=$(LOCAL_TEST_NAMESPACE) operator-sdk up local --namespace=$(APP_NAMESPACE) --verbose
```
Comment (alexeykazakov) [inline make/dev.mk]: It looks like `operator-sdk` is using `"-mod=vendor"`. So, we have to call `make vendor` here too:
```
Running []string{"go", "build", "-o", "build/_output/bin/member-operator-local", "-mod=vendor", "github.com/codeready-toolchain/member-operator/cmd/manager"}
```
Note: `make vendor` doesn't not exist in this repo yet. So we would need to copy it from https://github.com/codeready-toolchain/api/blob/8dbd7410a0532bcb4e6bcf2756460d37c7252e9a/make/go.mk#L22

File: README.adoc [hunk 2]
```diff
@@ -6,3 +6,59 @@ This is the CodeReady Toolchain Member Operator repository. It contains the Open
...
+To connect these two clusters together it is necessary to run a script link:scripts/add-cluster.sh[] that takes two arguments:
+
+1. type of the cluster to be added
+2. name of the cluster to be added
+
+The script is working with Minishift profiles and expects that there are two clusters (profiles) called `host` and `member`.
+To be able to run the script, you need to have both clusters prepared, which means:
+
+- created CRDs
+- created ServiceAccount with lusterRole/ClusterRoleBinding
```
Comment (alexeykazakov) [inline README.adoc]:
```suggestion
- created ServiceAccount with ClusterRole/ClusterRoleBinding
```

File: deploy/role.yaml [hunk 3]
```diff
@@ -39,6 +39,18 @@ rules:
...
+- apiGroups:
+  - core.kubefed.k8s.io
+  resources:
+  - KubeFedCluster
+  verbs:
+  - "*"
+- apiGroups:
+  - ""
+  resources:
+  - nodes
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply deploy/role.yaml]: Yes, the controller (apart from checking the health) also gets the kubernetes cluster zones and region by inspecting labels on nodes in the cluster, thus he needs to list them

File: deploy/role_binding.yaml [hunk 4]
```diff
@@ -1,11 +1,12 @@
-kind: RoleBinding
+kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: member-operator
 subjects:
 - kind: ServiceAccount
   name: member-operator
+  namespace: test-saas-control-plane
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply deploy/role_binding.yaml]: renamed e8ab41d

File: examples/kubefedcluster_host.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+apiVersion: core.kubefed.k8s.io/v1beta1
+kind: KubeFedCluster
+metadata:
+  name: host
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply examples/kubefedcluster_host.yaml]: Right, I wanted to add the labels later, but it makes more sense to add it now as part of the initial work. 65f96f2

File: make/dev.mk (hunk 1 above)
Comment (MatousJobanek) [reply make/dev.mk]: Correct, I had it already generated locally, so I didn't notice that it is needed :-/ 65573cd

File: scripts/add-cluster.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply scripts/add-cluster.sh]: Agree. There will be maybe more parts of the KubeFedCluster logic moved to the shared place, but the script is a good start: 7eeaa03

[the remark this answers is not in the document]
Comment (MatousJobanek) [reply scripts/add-cluster.sh]: and related PR in toolchain repo https://github.com/codeready-toolchain/toolchain/pull/2

**Pull request merged**